br/pkg/lightning/backend/local/duplicate_test.go

```diff
@@ -70,3 +73,178 @@ func TestBuildDupTask(t *testing.T) {
+	}
+	tctx := encoder.SessionCtx.GetTableCtx()
+	_, err = encoder.Table.AddRecord(tctx, data1)
+	require.NoError(t, err)
+	kvPairs := encoder.SessionCtx.TakeKvPairs()
+
+	data1RowKey := kvPairs.Pairs[0].Key
+	data1RowValue := kvPairs.Pairs[0].Val
+
+	originalErr := common.ErrFoundDuplicateKeys.FastGenByArgs(data1RowKey, data1RowValue)
+	rawKey, rawValue, err := local.RetrieveKeyAndValueFromErrFoundDuplicateKeys(originalErr)
+	require.NoError(t, err)
+	require.Equal(t, data1RowKey, rawKey)
+	require.Equal(t, data1RowValue, rawValue)
+}
+
+func TestConvertToErrFoundConflictRecords(t *testing.T) {
+	p := parser.New()
+	node, _, err := p.ParseSQL("create table a (a int primary key, b int not null, c text, unique key key_b(b));")
+	require.NoError(t, err)
+	mockSctx := mock.NewContext()
+	info, err := ddl.MockTableInfo(mockSctx, node[0].(*ast.CreateTableStmt), 108)
+	require.NoError(t, err)
+	info.State = model.StatePublic
+	tbl, err := tables.TableFromMeta(lkv.NewPanickingAllocators(0), info)
+	require.NoError(t, err)
+
+	sessionOpts := encode.SessionOptions{
+		SQLMode:   mysql.ModeStrictAllTables,
+		Timestamp: 1234567890,
+	}
+
+	encoder, err := lkv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+		Logger:         log.L(),
+	})
+	require.NoError(t, err)
+	encoder.SessionCtx.GetSessionVars().RowEncoder.Enable = true
+
+	data1 := []types.Datum{
+		types.NewIntDatum(1),
+		types.NewIntDatum(6),
+		types.NewStringDatum("1.csv"),
+	}
+	data2 := []types.Datum{
+		types.NewIntDatum(2),
+		types.NewIntDatum(6),
+		types.NewStringDatum("2.csv"),
+	}
+	data3 := []types.Datum{
+		types.NewIntDatum(3),
+		types.NewIntDatum(7),
+		types.NewStringDatum("3.csv"),
+	}
+	tctx := encoder.SessionCtx.GetTableCtx()
+	_, err = encoder.Table.AddRecord(tctx, data1)
+	require.NoError(t, err)
+	_, err = encoder.Table.AddRecord(tctx, data2)
+	require.NoError(t, err)
+	_, err = encoder.Table.AddRecord(tctx, data3)
+	require.NoError(t, err)
+	kvPairs := encoder.SessionCtx.TakeKvPairs()
+
+	data2RowKey := kvPairs.Pairs[2].Key
+	data2RowValue := kvPairs.Pairs[2].Val
+	data3IndexKey := kvPairs.Pairs[5].Key
+	data3IndexValue := kvPairs.Pairs[5].Val
+
+	originalErr := common.ErrFoundDuplicateKeys.FastGenByArgs(data2RowKey, data2RowValue)
+
+	newErr := local.ConvertToErrFoundConflictRecords(originalErr, tbl)
+	require.EqualError(t, newErr, "[Lightning:Restore:ErrFoundDataConflictRecords]found data conflict records in table a, primary key is '2', row data is '(2, 6, \"2.csv\")'")
+
+	originalErr = common.ErrFoundDuplicateKeys.FastGenByArgs(data3IndexKey, data3IndexValue)
+
+	newErr = local.ConvertToErrFoundConflictRecords(originalErr, tbl)
+	require.EqualError(t, newErr, "[Lightning:Restore:ErrFoundIndexConflictRecords]found index conflict records in table a, index name is 'a.key_b', unique key is '[7]', primary key is '3'")
+}
+
+func TestConvertToErrFoundConflictRecordsMultipleColumnsIndex(t *testing.T) {
```

Comment:
`TestConvertToErrFoundConflictRecords` and `TestConvertToErrFoundConflictRecordsMultipleColumnsIndex` only difference seems to be the number of columns in the table and the index. Could we unify it?

Reply:
The three unit tests have been unified by extracting `buildTableForTestConvertToErrFoundConflictRecords` function.